templates/templates_create.adoc

```diff
@@ -0,0 +1,223 @@
+
+[[edit-a-template]]
+== Creating a Template from Another Template
+
+You can create a new template based on one that has already been uploaded to your project: 
+
+The default templates are located in
+ifdef::openshift-origin[]
+*_/openshift-ansible/roles/openshift_examples/files/examples/v3.7/_*
+endif::[]
+ifdef::openshift-enterprise[]
+*_/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v3.7/_*
+endif::[]
+.
+
+. Make a copy of the template you want to edit:
++
+----
+cp <template-name>.yaml <new-template-name>.yaml
+cp <template-name>.json <new-template-name>.json
+----
++
+For example:
++
+----
+cp sso71-https.json sso72-https.json
+----
+
+. In a text editor, edit the fields as needed and save. Make sure to change the template name. 
+
+. Create the template object:
++
+----
+oc create -f <temaplte-name>.json
+----
++
+For example:
++
+----
+oc create -f sso72.json
+----
+
+You can now xref:../../dev_guide/templates/templates_using.adoc#dev-guide-templates-using[use the template to create objects] the same as any other template.
+
+[[export-as-template]]
+== Creating a Template from Existing Objects
+ifdef::openshift-online[]
+If you are upgrading from {product-title} Starter to {product-title} Pro, use
+`oc export all` to export all of your existing objects. {product-title} Pro does
+not support per-object resource migration.
+endif::[]
+
+Rather than writing an entire template from scratch, you can export existing
+objects from your project in template form, and then modify the template from
+there by adding parameters and other customizations. 
+
+To create a template from an object:
+
+. Export an object in a project in template form:
++
+----
+$ oc export <object-type> <object-name> --as-template=<template_name> > <path-to-template_filename>
+----
++
+The `<template-name>` is assigned to the `name` parameter in the template and 
+`<path-to-template_filename>` is the file name and location where the template is created.
++
+The `<object-type>` can be any of the following:
+
+- BuildConfig
+- Build
+- DeploymentConfig
+- ImageStream
+- Pod
+- ReplicationController
+- Route
+- Service
+- Template
++
+For example:
++
+----
+oc export pod mypod-1 --as-template pod-template > /etc/template/pod-template.yaml
+----
+
+. In a text editor, edit the fields as needed and save. Make sure to change the template name and 
+add a description if not present. 
+
+. Create the template object:
++
+----
+oc create -f <temaplte-name>.yaml
+----
++
+For example:
++
+----
+oc create -f pod-template.yaml
+----
+
+You can now xref:../../dev_guide/templates/templates_using.adoc#dev-guide-templates-using[use the template to create objects] the same as any other template.
+
+[[other-recommendations]]
+== Template Creation Best Practices
+
+* Group related services together in the management console by adding the
+`service.alpha.openshift.io/dependencies` annotation to the Service object in
+your template.
```

Comment:
This annotation is no longer used. (The overview has been redesigned since then.)